src/pubsub/ua_pubsub_writergroup.c

```diff
@@ -1010,9 +1070,16 @@ sendNetworkMessageUADP(UA_PubSubConnection *connection, UA_WriterGroup *wg,
     rv = encodeNetworkMessage(wg, &nm, &buf);
     UA_CHECK_STATUS(rv, goto cleanup_with_msg_size);
 
+    /* Choose the channel */
+    UA_PubSubChannel *channel = NULL;
+    if(transportSettings->content.decoded.data) {
+        channel = wg->channel;
+    } else {
+        channel = connection->channel;
+    }
```

Comment:
This can result in a SegFault when the transportSettings->type is DatagramWriterGroupTransportDataType (not DatagramWriterGroupTransport2DataType). This can happen when configuring a multicast publisher and setting the transportsettings